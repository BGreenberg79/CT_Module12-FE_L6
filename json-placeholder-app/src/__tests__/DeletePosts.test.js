import React from 'react';
import axios from 'axios';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import PostList from '../components/PostList';
import { MemoryRouter } from 'react-router-dom';
import { act } from 'react';
import '@testing-library/jest-dom';

jest.mock('axios');


test('Delete button is clicked and sends a delete request', async () => {


    // Mock the axios.get response to return posts
    axios.get.mockResolvedValueOnce({
        data: [{ id: 1, title: 'Sample Title', body: 'Sample Body', userId: 1 }],
    });

    axios.delete.mockResolvedValueOnce({
        status: 200,
    })

    await act(async () =>{
        render(
            <MemoryRouter>
                <PostList />
            </MemoryRouter>
        )
    });

    // Ensure the post is rendered first
    await waitFor(() => {
    expect(screen.getByText(/Sample Title/i)).toBeInTheDocument();
    expect(screen.getByText(/Sample Body/i)).toBeInTheDocument();
    });

    // Find delete button and click
    const button = await screen.findByRole('button',{ name: /Delete/i } )
    expect(button).toBeInTheDocument();
    fireEvent.click(button);


    // Assert axios.delete was called correctly
    await waitFor(() =>
        expect(axios.delete).toHaveBeenCalledWith(
            'https://jsonplaceholder.typicode.com/posts/1'
        )
    );
});


test('matches the snapshot', async () => {

    axios.get.mockResolvedValueOnce({
        data: [{ id: 1, title: 'Sample Title', body: 'Sample Body', userId: 1 }],
    });

    const { asFragment } = render(
                                <MemoryRouter>
                                    <PostList />
                                </MemoryRouter>
                            );
    
    await waitFor(() => {
        expect(screen.getByText(/Sample Title/i)).toBeInTheDocument();
        expect(screen.getByText(/Sample Body/i)).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
});