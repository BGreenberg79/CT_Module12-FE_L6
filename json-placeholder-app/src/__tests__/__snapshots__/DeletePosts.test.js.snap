// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h2>
      JSON Placeholder Post List
    </h2>
    <div
      class="list-group"
    >
      <div
        class="list-group-item list-group-item-info"
      >
        Post ID: 1
        <br />
        User ID: 1
        <br />
        Title: Sample Title
        <br />
        Body: Sample Body
        <button
          class="shadow-sm m-1 p-1 btn btn-warning"
          type="button"
        >
          Edit
        </button>
        <button
          class="shadow-sm m-1 p-1 btn btn-danger"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
